### PR TITLE
add test to check prometheus scraping loki through traefik

### PR DESCRIPTION
**Pull request by lucabello**
## Issue
This addresses the first part of https://github.com/canonical/loki-k8s-operator/issues/211.

## Release Notes
This PR adds an integration test to make sure Prometheus is correctly scraping Loki when Loki is behind traefik.

### CODE REVIEW

File: tests/integration/test_bundle.py [hunk 1]
```diff
@@ -196,3 +199,33 @@ async def test_bundle_charms_can_handle_frequent_update_status(ops_test: OpsTest
...
+async def test_prometheus_scrapes_loki_through_traefik(ops_test: OpsTest):
+    """Prometheus should correctly scrape Loki through its traefik endpoint."""
+    prom_url = await get_proxied_unit_url(ops_test, "prometheus", 0)
+
+    response = urllib.request.urlopen(f"{prom_url}/api/v1/targets", data=None, timeout=2.0)
+    assert response.code == 200
+    targets = json.loads(response.read())["data"]["activeTargets"]
+    targets_summary = [(t["discoveredLabels"]["__metrics_path__"], t["health"]) for t in targets]
+    assert (f"/{ops_test.model.name}-loki-0/metrics", "up") in targets_summary
+    logger.info("prometheus is successfully scraping loki through traefik")
```
Comment (sed-i) [inline tests/integration/test_bundle.py:213]: Could this be [simplified](https://github.com/canonical/prometheus-k8s-operator/blob/1a25f576ed3afb5965a44cdaaa2c768d0dcfa00a/tests/integration/test_external_url.py#L190) by using a regular string search?
```
response = response.read()
assert '"health":"up"' in response
assert '"health":"down"' not in response
```

File: tests/integration/test_bundle.py [hunk 2]
```diff
@@ -196,3 +199,33 @@ async def test_bundle_charms_can_handle_frequent_update_status(ops_test: OpsTest
...
+    # Create the relation
+    await ops_test.model.add_relation("loki", "zinc")
+    await ops_test.model.wait_for_idle(status="active")
+    # Check that logs are coming in
+    response = urllib.request.urlopen(f"{loki_url}/loki/api/v1/series")
+    assert response.code == 200
+    series = json.loads(response.read())["data"]
+    series_charms = [s["juju_charm"] for s in series]
+    assert "zinc-k8s" in series_charms
+    logger.info("loki is successfully receiving zinc logs through traefik")
```
Comment (sed-i) [inline tests/integration/test_bundle.py:231]: Is this `info` a leftover from debugging?

**Review by sed-i — APPROVED** [review]

File: tests/integration/test_bundle.py (hunk 1 above)
Comment (lucabello) [reply tests/integration/test_bundle.py:213]: Not really, as the response contains the status for all the targets; checking that everything is up would misrepresent the test which is only checking for loki

File: tests/integration/test_bundle.py (hunk 2 above)
Comment (lucabello) [reply tests/integration/test_bundle.py:231]: Nope, it mimics what's done in all the other tests!

**Pull request merged**